mne/morph.py

```diff
@@ -0,0 +1,731 @@
+# Author(s): Tommy Clausner <tommy.clausner@gmail.com>
+
+# License: BSD (3-clause)
+
+
+import os.path as op
+import warnings
+
+import numpy as np
+from scipy import sparse
+
+from . import read_forward_solution, compute_morph_matrix
+from .externals.h5io import read_hdf5, write_hdf5
+from .externals.six import string_types
+from .source_estimate import (VolSourceEstimate, SourceEstimate,
+                              VectorSourceEstimate, _get_subject_sphere_tris)
+from .source_space import SourceSpaces
+from .utils import logger, verbose, check_version
+
+
+class SourceMorph(object):
+    """Container for source estimate morphs.
+
+    Parameters
+    ----------
+    src : instance of SourceSpaces
+        The list of SourceSpaces corresponding subject_from
+    subject_from : str | None
+        Name of the original subject as named in the SUBJECTS_DIR.
+        If None src[0]['subject_his_id]' will be used. The default is None.
+    subject_to : str | array | list of two arrays
+        Name of the subject on which to morph as named in the SUBJECTS_DIR
+        The default is 'fsaverage'. If morphing a surface source extimate,
+        subject_to can also be an array of vertices or a list of two arrays of
+        vertices to morph to. If morphing a volume source space, subject_to can
+        be the path to a MRI volume.
+    subjects_dir : str | None
+        Path to SUBJECTS_DIR if it is not set in the environment. The default
+        is None.
+    verbose : bool | str | int | None
+        If not None, override default verbose level (see :func:`mne.verbose`
+        and :ref:`Logging documentation <tut_logging>` for more).
+
+    Surface / Vector Source Estimate parameters
+    -------------------------------------------
+    smooth : int | None
+        Number of iterations for the smoothing of the surface data.
+        If None, smooth is automatically defined to fill the surface
+        with non-zero values.
+    warn : bool
+        If True, warn if not all vertices were used.
+    xhemi : bool
+        Morph across hemisphere. Currently only implemented for
+        ``subject_to == subject_from``. See notes below.
+
+
+    Volume Source Estimate parameters
+    ---------------------------------
+    niter_affine : tuple of int
+        Number of levels (``niter_affine.__len__()``) and number per level of
+        iterations to perform the affine transform. Increasing index values
+        for the tuple mean later levels and each int represents the number
+        of iterations in that level. Default is niter_affine=(100, 100, 10)
+    niter_sdr : tuple of int
+        Number of levels (``niter_sdr.__len__()``) and number per level of
```

Comment:
same comment here

Reply:
also I would decompress the docstring a bit. For example, even after reading the docstring twice, I am not sure what is a "level"